ui/src/app/workflows/components/submit-workflow-panel.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from 'react';
+import {Parameter, Workflow} from '../../../models';
+import {uiUrl} from '../../shared/base';
+import {ErrorNotice} from '../../shared/components/error-notice';
+import {services} from '../../shared/services';
+
+import {Select} from 'argo-ui';
+import {TagsInput} from '../../shared/components/tags-input/tags-input';
+
+interface Props {
+    kind: string;
+    namespace: string;
+    name: string;
+    entrypoint: string;
+    entrypoints: string[];
+    parameters: Parameter[];
+}
+
+interface State {
+    entrypoint: string;
+    parameters: Parameter[];
+    labels: string[];
+    error?: Error;
+}
+
+export class SubmitWorkflowPanel extends React.Component<Props, State> {
+    constructor(props: any) {
+        super(props);
+        this.state = {
+            entrypoint: this.props.entrypoint || (this.props.entrypoints.length > 0 && this.props.entrypoints[0]),
+            parameters: this.props.parameters,
+            labels: ['submit-from-ui=true']
+        };
+    }
+
+    public render() {
+        return (
+            <>
+                <h4>Submit Workflow</h4>
```

Comment:
Should this be parametrized? `Submit CronWorkflow`, `Submit WorkflowTempalte`, etc.

Reply:
maybe, but I merged